app/src/pages/prompt/PromptChatMessages.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useMemo } from "react";
+import { useFragment } from "react-relay";
+import { graphql } from "relay-runtime";
+
+import { Flex, Text } from "@arizeai/components";
+
+import { TemplateLanguages } from "@phoenix/components/templateEditor/constants";
+import { TemplateLanguage } from "@phoenix/components/templateEditor/types";
+
+import {
+  PromptChatMessages__main$key,
+  PromptTemplateFormat,
+} from "./__generated__/PromptChatMessages__main.graphql";
+import { ChatTemplateMessage } from "./ChatTemplateMessage";
+import { PromptChatTemplate, PromptChatTemplateSchema } from "./schemas";
+
+const convertTemplateFormat = (
+  templateFormat: PromptTemplateFormat
+): TemplateLanguage => {
+  if (templateFormat === "FSTRING") {
+    return TemplateLanguages.FString;
+  } else if (templateFormat === "MUSTACHE") {
+    return TemplateLanguages.Mustache;
+  }
+  return TemplateLanguages.NONE;
+};
+
+export function PromptChatMessages({
+  promptVersion,
+}: {
+  promptVersion: PromptChatMessages__main$key;
+}) {
+  const { template, templateType, templateFormat } = useFragment(
+    graphql`
+      fragment PromptChatMessages__main on PromptVersion {
+        template
+        templateType
+        templateFormat
+      }
+    `,
+    promptVersion
+  );
+
+  if (templateType === "STRING") {
+    return <Text>{template}</Text>;
+  }
+
+  return (
+    <ChatMessages
+      template={template}
+      templateFormat={convertTemplateFormat(templateFormat)}
+    />
+  );
+}
+
+function ChatMessages({
+  template,
+  templateFormat,
+}: {
+  template: PromptChatTemplate | unknown;
+  templateFormat: TemplateLanguage;
+}) {
+  const messages = useMemo(() => {
+    const parsedTemplate = PromptChatTemplateSchema.safeParse(template);
+    if (!parsedTemplate.success) {
+      return [];
```

Comment:
thought: display an error if you fail to parse? I guess this code will go away

Reply:
Yeah this is intended to be deleted next PR ideally